Source/Core/DolphinWX/Cheats/CheatSearchTab.cpp

```diff
@@ -251,15 +251,33 @@ void CheatSearchTab::UpdateCheatSearchResultItem(long index)
 	m_lview_search_results->SetItem(index, 3, buf);
 }
 
+enum class ComparisonMask
+{
+	EQUAL        = 0x1,
+	GREATER_THAN = 0x2,
+	LESS_THAN    = 0x4
+};
+
+static ComparisonMask operator | (ComparisonMask comp1, ComparisonMask comp2)
+{
+	return static_cast<ComparisonMask>(static_cast<int>(comp1) |
+					   static_cast<int>(comp2));
+}
+
+static ComparisonMask operator & (ComparisonMask comp1, ComparisonMask comp2)
+{
+	return static_cast<ComparisonMask>(static_cast<int>(comp1) &
+					   static_cast<int>(comp2));
+}
+
 void CheatSearchTab::FilterCheatSearchResults(u32 value)
 {
-	// Determine the selected filter
-	// 1 : equal
-	// 2 : greater-than
-	// 4 : less-than
-	// 6 : not equal
-	static const int filters[] = { 6, 1, 2, 4 };
-	int filter_mask = filters[m_search_type->GetSelection()];
+	static const ComparisonMask filters[] = { ComparisonMask::EQUAL | ComparisonMask::GREATER_THAN | ComparisonMask::LESS_THAN, // Unknown
```

Comment:
static const std::array<ComparisonMask, 5> filters{
    ComparisonMask::EQUAL | ComparisonMask::GREATER_THAN | ComparisonMask::LESS_THAN, // Unknown
    ComparisonMask::GREATER_THAN | ComparisonMask::LESS_THAN, // Not Equal
    ComparisonMask::EQUAL,
    ComparisonMask::GREATER_THAN,
    ComparisonMask::LESS_THAN
};